game/sourcemod/scripting/donationcontrol.sp

```diff
@@ -4,187 +4,183 @@
+bool cv_bEnabled, cv_bFullScreen, cv_bAdvertStatus;
+char cv_sURL[255];
+float cv_fAdvertTime;
+int cv_iMinimumAmount;
 
-new Handle:hDisplayMenu = INVALID_HANDLE;
-new Handle:hAdvertTimer = INVALID_HANDLE;
+Menu hDisplayMenu;
+Handle hAdvertTimer;
 
-public Plugin:myinfo =
-{
-	name = PLUGIN_NAME,
-	author = "Keith Warren (Drixevel) & NineteenEleven",
-	description = "Allows clients to donate via the Donations Control Panel script in-game.",
-	version = PLUGIN_VERSION,
-	url = "http://www.drixevel.com/"
-}
+public Plugin myinfo = {
+	name 		= PLUGIN_NAME,
+	author 		= "Keith Warren (Drixevel), NineteenEleven & Sgt. Gremulock",
+	description = "Allows clients to donate via the Donations Control Panel script in-game (syntax updated by Sgt. Gremulock).",
+	version 	= PLUGIN_VERSION,
+	url 		= "http://www.drixevel.com/"
+};
 
-public OnPluginStart()
+public void OnPluginStart()
 {
 	LoadTranslations("common.phrases");
 	LoadTranslations("donationcontrol.phrases");
 
-	CreateConVar("donations_control_version", PLUGIN_VERSION, PLUGIN_NAME, FCVAR_REPLICATED|FCVAR_NOTIFY|FCVAR_PLUGIN|FCVAR_SPONLY|FCVAR_DONTRECORD);
-	hConVars[0] = CreateConVar("sm_kpanel_enable", "1", "Enable or disable plugin", 0, true, 0.0, true, 1.0);
+	CreateConVar("donations_control_version", PLUGIN_VERSION, PLUGIN_NAME, FCVAR_REPLICATED|FCVAR_NOTIFY|FCVAR_DONTRECORD);
+	hConVars[0] = CreateConVar("sm_kpanel_enable", "1", "Enable or disable plugin", _, true, 0.0, true, 1.0);
 	hConVars[1] = CreateConVar("sm_kpanel_url", "https://website.com/donations", "URL to your Donations Control installation.");
-	hConVars[2] = CreateConVar("sm_kpanel_fullscreen", "1", "Enable or disable fullscreen windows", 0, true, 0.0, true, 1.0);
-	hConVars[3] = CreateConVar("sm_kpanel_advertisement", "1", "Display plugin creator advertisement: (1 = on, 0 = off)", FCVAR_PLUGIN, true, 0.0, true, 1.0);
-	hConVars[4] = CreateConVar("sm_kpanel_advertisement_time", "120.0", "Timer between messages: (1.0 + )", FCVAR_PLUGIN, true, 1.0);
-	hConVars[5] = CreateConVar("sm_kpanel_minimum", "4", "Minimum amount to donate: (Default: 4, Less than or equal to)", FCVAR_PLUGIN, true, 1.0);
+	hConVars[2] = CreateConVar("sm_kpanel_fullscreen", "1", "Enable or disable fullscreen windows", _, true, 0.0, true, 1.0);
+	hConVars[3] = CreateConVar("sm_kpanel_advertisement", "1", "Display plugin creator advertisement: (1 = on, 0 = off)", _, true, 0.0, true, 1.0);
+	hConVars[4] = CreateConVar("sm_kpanel_advertisement_time", "120.0", "Timer between messages: (1.0 + )", _, true, 1.0);
+	hConVars[5] = CreateConVar("sm_kpanel_minimum", "4", "Minimum amount to donate: (Default: 4, Less than or equal to)", _, true, 1.0);
 
-	for (new i = 0; i < sizeof(hConVars); i++)
+	for (int i = 0; i < sizeof(hConVars); i++)
 	{
-		HookConVarChange(hConVars[i], HandleCvars);
+		hConVars[i].AddChangeHook(HandleCvars);
 	}
 
 	RegConsoleCmd("sm_donate", DonatePanel);
 
-	//AutoExecConfig(true, "plugin.donation.control");
+	AutoExecConfig();
```

Comment:
`autoCreate` and `name` parameter should be passed here?

Also, any idea why was it commented out previously?

Reply:
The parameters by default for AutoExecConfig pass true for the first parameter, and the name of the config file defaults to "plugin.donationcontrol.cfg". Therefore, setting those 2 parameters is unnecessary.